spec/features/course/achievement_management_spec.rb

```diff
@@ -1,136 +1,202 @@
+          select_option
+        click_button I18n.t('helpers.submit.condition_achievement.update')
+        expect(current_path).to eq edit_course_achievement_path(course, achievement)
+        expect(page).to have_selector('tr.condition > td:nth-child(2)',
+                                      text: achievement_to_change_to.title)
+      end
 
-      it 'deletes the achievement' do
+      scenario 'I can delete an achievement condition' do
+        achievement_condition, achievement, other_achievement =
+          create_achievement_condition
+        visit edit_course_achievement_path(course, achievement)
+        condition_delete_path =
+          course_achievement_condition_achievement_path(course, achievement,
+                                                        achievement_condition)
         expect do
-          # Show and delete have the same URL, difference is in the request method
-          # find the 2nd link that matches the path [2], this is the delete button
-          find_link(nil, href: achievement_path, between: 2..2).click
-        end.to change(course.achievements, :count).by(-1)
+          find_link(nil, href: condition_delete_path).click
+        end.to change { achievement.conditions.count }.by(-1)
+        expect(page).not_to have_selector('tr.condition > td:nth-child(2)',
+                                          text: other_achievement.title)
       end
 
-      context 'after achievement deleted' do
-        before { find_link(nil, href: achievement_path, between: 2..2).click }
+      scenario 'I can create a level condition' do
+        achievement = create(:course_achievement, course: course)
+        visit edit_course_achievement_path(course, achievement)
+        expect(page).to have_selector(:link_or_button,
+                                      I18n.t('course.condition.levels.new.header'))
+        click_link I18n.t('course.condition.levels.new.header')
+        expect(current_path).to eq(new_course_achievement_condition_level_path(course,
+                                                                               achievement))
+        fill_in 'minimum_level', with: '10'
+        click_button I18n.t('helpers.submit.condition_level.create')
+        expect(current_path).to eq edit_course_achievement_path(course, achievement)
+        expect(page).to have_selector('tr.condition > td:nth-child(2)', text: '10')
+      end
 
-        it 'shows the success message' do
-          expect(page).to have_selector('div',
-                                        text: I18n.t('course.achievements.destroy.success'))
-        end
+      scenario 'I can edit a level condition' do
+        level_condition, achievement = create_level_condition
+        visit edit_course_achievement_path(course, achievement)
+        expect(current_path).to eq edit_course_achievement_path(course, achievement)
+        expect(page).to have_selector('tr.condition > td:nth-child(2)', text: '1')
```

Comment:
why are you using nth-child? you should give the td a class and search for the class.